### ISSUE TITLE
Objects added to an event in somewhat random order

### ISSUE THREAD

**Issue by automationator** [MISP/MISP#6645]
I am using MISP 2.4.134 and am seeing weird behavior with how objects are added to an event when done in bulk.

Specifically, I'm using PyMISP (same version) and am appending new objects to the MISPEvent's objects list. Then I am using the PyMISP update_event function. This uploads all of my objects to the event in _mostly_ the same order they're in the objects list.

However, when you look at the event in the MISP web console, a handful of the objects appear in a seemingly random order and do not match the order they're in inside of the MISPEvent objects list. If I delete all of the objects from the event and try my update_event API call again, the order is also not consistent with the previous API call. I have logged the contents of the objects list as I make this API call so that I can compare it with how they are presented in the web console, and they never match completely.

Yet if I make individual API calls to add each object to the event separately using the PyMISP add_object function, the order is as I would expect and is consistent every single time.

Some of the objects I am uploading to my event are file objects that are whitelisted due to their filepath attributes. I would not expect this to affect order of the rest of the valid objects, but I figured I should mention that as well.

**Comment by mokaddem** [comment]
Hey,
By order, do you mean the order on which elements are presented in the attributes/objects table under `events/view/[id]`?
In that case, the order is only based on the timestamp of these elements, which is the time at which they were last updated.
By adding them one by one, they will have a different timestamp resulting in the order you would expect.
Cheers

**Comment by automationator** [comment]
Yes, that's what I'm referring to. That's a shame that's how it works since it's significantly slower to add the objects one by one. Thanks for the quick reply.

**Comment by mokaddem** [comment]
I'm not sure to understand why the order is important. I don't know if it's applicable for your use case, but setting a value for the `first_seen` and sorting the index on that could help.

**Comment by automationator** [comment]
The order of the objects is very important. A lot of our MISP events are created automatically, and we rely on the order of the objects to help tell the story of the attack.

For example, let’s say the MISP event contains a phishing email. That email object would come first. If the email had an attachment, we’d want that file object immediately after the email object. If that email attachment ran through our sandboxes, we’d want various objects like http requests or other file objects for files the sample dropped on the system during execution to be next in the list, and so on. 

Being able to control the order like this lets you scroll down the MISP event and see the entire chain of events. If the objects get jumbled up and out of order, it’s much harder to do and requires you to do a lot more manual work to piece together that “parent child” type of object relationship.

**Comment by iglocska** [comment]
The order will not be easy to control, especially once you take synchronisation into account. 

However, for telling a story, you can add relationships between objects, it's MUCH preferred to time-based ordering. This also allows you to graph your story directly in MISP, such as this:

![image](https://user-images.githubusercontent.com/3668672/100325426-e82a8f00-2fc8-11eb-9005-4b477346289f.png)

**Comment by automationator** [comment]
The order is simple to control if you use the add_object function one by one instead. However, the object relationship is interesting. I will give that a look.

**Comment by mokaddem** [comment]
Also setting the `first_seen` on your objects/attributes can be used for such purpose.
If set, you can use the *event timeline* in `/events/view` to see a visual reprenstation.
